erpnext/accounts/doctype/pricing_rule/pricing_rule.py

```diff
@@ -175,6 +175,51 @@ def apply_pricing_rule(args, doc=None):
 
 	return out
 
+@frappe.whitelist()
+def apply_transaction_pricing_rule(doc):
+	from erpnext.accounts.doctype.pricing_rule.utils import apply_pricing_rule_on_transaction
+
+	# converting dict into doc
+	if isinstance(doc, string_types):
+		doc = json.loads(doc)
+	if doc:
```

Comment:
I don't know if this condition is needed at all. And if it _is_ needed, what happens if it's falsy?